src/components/Notifications/NotificationWrapper.tsx

```diff
@@ -0,0 +1,91 @@
+import React from 'react';
+
+import {useMobile} from '@gravity-ui/uikit';
+
+import {Notification} from '../Notification';
+import {NotificationWithSwipe} from '../Notification/NotificationWithSwipe';
+import {NotificationProps} from '../Notification/definitions';
+import {block} from '../utils/cn';
+
+import './Notifications.scss';
+
+const b = block('notifications');
+
+export const NotificationWrapper = (props: {
+    notification: NotificationProps;
+    swipeThreshold?: number;
+}) => {
+    const ref = React.useRef<HTMLDivElement>(null);
+
+    const {notification, swipeThreshold} = props;
+    const [mobile] = useMobile();
+    const [wrapperMaxHeight, setWrapperMaxHeight] = React.useState<number | undefined>(undefined);
+    const [isRemoved, setIsRemoved] = React.useState(false);
+
+    React.useEffect(() => {
```

Comment:
it is a little bit weird that notification manages it's visibility state by itself

Reply:
It's because we need to animate the removing effect (`archive`)
I thought it would be more convenient if we manage it in the `NotificationWrapper` component